### DISCUSSION TITLE
Retrieve credentials from Thycotic Secret server using power shell scripts.

### DISCUSSION THREAD

**Discussion by amarreddy86** [thycotic-ps/thycotic.secretserver#275] [General]
Team, Does any one has script to "Retrieve credentials from Secret server using power shell scripts. "

**Comment by jagger** [comment]
I would use new-tsssession to get an auth token, then use that with get-tsssecret to retrieve the contents of the secret

**Reply by amarreddy86** [reply]
Hi @jagger , Thanks for your quick response. if possible can you please put the full script . Am just new to this . Thanks.

**Comment by AngelNev** [comment]
from: https://thycotic-ps.github.io/thycotic.secretserver/commands/secrets/Get-TssSecret.html

$ssCred = Get-Credential
$session = New-TssSession -SecretServer https://alpha -Credential $ssCred
Get-TssSecret -TssSession $session -Id 93

**Reply by amarreddy86** [reply]
Hi @jagger  it does not get the password of the secret. i am looking for a script which will retrieve the password . 

i tried this but unfortunately no luck to see the password .  

It helps the administrators to pickup passwords easily.

**Reply by enphyniti** [reply]
Change that last line to this:
$myCred = ((Get-TssSecret -TssSession $session -SecretId 93).GetCredential('domain','username','password'))

That will create a credential you can use in your script.  Or you can display it like this:
$myCred.GetNetworkCredential().password
